### ISSUE TITLE
standard_basis for modules accepts monomial ordering, but silently ignores it

### ISSUE THREAD

**Issue by afkafkafk13** [oscar-system/Oscar.jl#2155]
standard_basis accepts a keyword argument ordering containing a module ordering (even based on a local or mixed ring ordering), but ignores it

fragment of Oscar session:
```
julia> r
Multivariate Polynomial Ring in y, z over Rational Field

julia> o1
negdegrevlex([y, z])*lex([gen(1), gen(2), gen(3)])

julia> Jm3
Submodule with 2 generators
1 -> 2*e[1] - 3*y^2*e[2] - 6*z^2*e[3]
2 -> 2*e[1] + 3*y^2*e[2] - 3*z^2*e[3]

julia> standard_basis(Jm3;ordering=o1)
Gröbner basis with elements
-4*e[1] + 9*z^2*e[3]
2*e[1] + 3*y^2*e[2] - 3*z^2*e[3]
 with respect to the ordering
negdegrevlex([y, z])*lex([gen(1), gen(2), gen(3)])
```

Explanation:
As the first comparison is on some negative weighted degree ordering, the leading monomial of each of the two generators is 2*e[1] and the first spoly should give rise to somthing with leading monomial y^2*e[2]. The above output shows clearly that this is not the case. Obviously some global ordering had been applied.

**System (please complete the following information):**
```
julia> Oscar.versioninfo(full=true)
OSCAR version 0.12.0-DEV - #notions_of_transforms, d77776aa -- 2023-03-27 09:27:14 +0200
  combining:
    AbstractAlgebra.jl   v0.28.2
    GAP.jl               v0.9.4
    Hecke.jl             v0.18.3
    Nemo.jl              v0.33.1
    Polymake.jl          v0.9.0
    Singular.jl          v0.17.0
  building on:
    Antic_jll               v0.201.500+0
    Arb_jll                 v200.2300.0+0
    Calcium_jll             v0.401.100+0
    FLINT_jll               v200.900.4+0
    GAP_jll                 v400.1200.200+0
    Singular_jll            v403.101.500+0
    libpolymake_julia_jll   v0.9.0+0
    libsingular_julia_jll   v0.29.0+0
    polymake_jll            v400.900.0+0
See `]st -m` for a full list of dependencies.

Julia Version 1.7.2
Commit bf53498635 (2022-02-06 15:21 UTC)
Platform Info:
  OS: Linux (x86_64-pc-linux-gnu)
  CPU: AMD Ryzen 5 PRO 3500U w/ Radeon Vega Mobile Gfx
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-12.0.1 (ORCJIT, znver1)
Official https://julialang.org/ release
...
```

**Additional context**
The method of the same name for MPolyIdeals works properly.

**Comment by ederc** [comment]
The resulting basis is w.r.t. `(dp(2), C)` in Singular notation. I _assume_ that the problem comes from `singular_assure(F::ModuleGens)` internally: Even so the corresponding `ModuleGens` element for `Jm3` has the correct `ordering`  set, the `singular_assure` seems to be _not_ taking care of any ordering at all. Thus it generates the Singular side with the standard Singular module ordering, which is, in this case, `(dp(2), C)`.

**Comment by HechtiDerLachs** [comment]
This seems to entail other issues:
```

julia> R, (x, y) = QQ["x", "y"]
(Multivariate polynomial ring in 2 variables over QQ, QQMPolyRingElem[x, y])

julia> F2 = free_module(R, 2)
Free module of rank 2 over Multivariate polynomial ring in 2 variables over QQ

julia> oo = lex(gens(F2))*negdegrevlex(gens(R))
lex([gen(1), gen(2)])*negdegrevlex([x, y])

julia> I, inc = sub(F2, [x*(x+1)*F2[1]])
(Submodule with 1 generator
1 -> (x^2 + x)*e[1]
represented as subquotient with no relations., Map with following data
Domain:
=======
Submodule with 1 generator
1 -> (x^2 + x)*e[1]
represented as subquotient with no relations.
Codomain:
=========
Free module of rank 2 over Multivariate polynomial ring in 2 variables over QQ
)

julia> leading_module(I, oo)
Submodule with 1 generator
1 -> x^2*e[1]
represented as subquotient with no relations.

julia> 
```
More than two months later and still the issue persists?

**Comment by afkafkafk13** [comment]
resolved by PR #2468
